### PR TITLE
Phase 0 — Foundations &amp; UX Overhaul (v0.7.x): complete §8 correctness backlog + pre-Phase-2 bug fixes

### PR DESCRIPTION
**Pull request by bberardi**
## Summary

This PR lands **Phase 0 — Foundations & UX Overhaul** (ROADMAP §5) in full, and completes the remaining correctness work that the Math Correctness Charter (§4) requires **before Phase 2** puts the numbers on charts.

Phase 0 items **0.1–0.12** were built up incrementally on this branch (CI workflow, date-indexed forecast engine, derived-data stripping + export schema v2, state context, dependency modernization to React 19 / MUI 9 / Vite 8, theme + dark mode, dialogs, validation UX, empty states, layout polish, the math verification suite, and the correctness regression sweep). The commits in this session close out the leftover **§8 Correctness Backlog** and the open math/validation bugs blocking Phase 2.

### ROADMAP §8.1 — reconcile the step-up anniversary off-by-one ✅
The two investment engines agreed to the cent **without** yearly step-ups but diverged **with** one: the monthly-grid `forecastInvestment` stepped up one contribution earlier than the period-indexed `generateInvestmentGrowth`. The period engine is canonical (it backs the PIT view and Growth Schedule popout), so `forecastInvestment` now attributes each grid-month contribution to the period-opening contribution one interval earlier (`getInvestmentYear(monthDate − interval)`). The engines now agree to the cent at every compounding boundary **with or without** step-ups.
- Flipped the `it.fails` tripwire in `forecast-consistency.test.ts` to a passing `it`.
- Added a hand-derived step-up **reference oracle** (Charter §4 layer 1) in `math-reference.test.ts`.
- Updated `PRECISION.md` §4 and `ROADMAP` §8.1 (marked resolved).

### Bug #73 — `getPitCalculation` RemainingTerms
A paid-off (early-payoff) loan reported `RemainingTerms = scheduled − payoff term` next to `RemainingPrincipal = $0` — internally contradictory. Now reports `0` when the loan is paid off. Regression assertion added.

### Bug #70 — under-amortizing loan (payment < interest)
`generateAmortizationSchedule` emitted negative-principal, balance-growing rows closing with a catastrophic balloon. It now stops instead of emitting that garbage, and `validateLoan` adds a matching **non-blocking sanity warning**. Edge-case test updated.

### Bug #72 — form validation stricter than the import path/engine
The form rejected `InterestRate = 0` (interest-free loans) and `StartingBalance = 0` ($0 new accounts) that the JSON import boundary and the engine both accept — trapping imported entities as uneditable. Form validation now accepts `>= 0` for both (still rejecting negatives), plus an empty-investment warning. Over-restriction assertions updated.

### ROADMAP §8.3 — Stryker mutation ratchet
Strengthened the validation warning tests to assert message **content** (not just presence), which restored the score after the new warning/guard code (`validation-helpers` 82.7 → 83.7%, overall **85.04%**). Ratcheted `thresholds.break` **80 → 83**, refreshed the baseline comment, and recorded `format-helpers`' 4 `Intl.NumberFormat`-cache survivors as an explicit equivalent-mutant waiver.

> Deferred (per agreed scope): §8.2 (migrate date math to dayjs) is tagged for Phase 6.

## Verification
- ✅ **301 tests pass** (`npm test`)
- ✅ **100% line + branch coverage** on `src/helpers/**` (CI coverage gate, `test:coverage`)
- ✅ Lint, format, build clean
- ✅ Stryker overall **85.04%** ≥ break threshold **83**

🤖 Generated with [Claude Code](https://claude.ai/code)

---
_Generated by [Claude Code](https://claude.ai/code/session_017StzahsGp1PDCTntGuQvhk)_

### CODE REVIEW

File: src/helpers/validation-helpers.ts [hunk 1]
```diff
@@ -113,6 +117,20 @@ export const validateLoan = (loan: Loan): ValidationResult<LoanField> => {
...
   }
+  // A payment that does not cover the first period's interest never amortizes
+  // the loan — the balance grows forever and the amortization schedule has
+  // nothing sensible to show. Warn (don't block: the value is type-valid) so
+  // the user sees why their projection looks wrong. (#70)
+  if (
+    loan.Principal > 0 &&
+    loan.InterestRate > 0 &&
+    typeof loan.MonthlyPayment === 'number' &&
+    loan.MonthlyPayment > 0 &&
```
Comment (bberardi) [inline src/helpers/validation-helpers.ts:132]: Boundary mismatch with the #70 schedule guard. This warning uses strict `<`, but `generateAmortizationSchedule` treats `normalPrincipal <= 0` as unamortizable and `break`s. At exact break-even (e.g. 100k @ 12%, payment = $1000 → interest $1000, `normalPrincipal === 0`), the schedule returns `[]` (it genuinely never pays off) yet no warning fires — so that loan silently produces an empty/zero projection with no explanation. The `breakEven` test in this PR even pins the no-warning behavior.

Suggest changing this condition to `<=` so the two boundaries agree:

```js
loan.MonthlyPayment <= (loan.Principal * loan.InterestRate) / 100 / 12
```

(and updating the `breakEven` test to expect the warning).

---
_Generated by [Claude Code](https://claude.ai/code)_

Comment (bberardi) [reply src/helpers/validation-helpers.ts:132]: Good catch — fixed in eacfdf6. The warning now uses `<=` to match `generateAmortizationSchedule`'s `normalPrincipal <= 0` guard, so exact break-even (payment == first-month interest) is explained instead of silently producing an empty schedule. Message reworded to "does not exceed the first month's interest", and the boundary tests updated: break-even now warns, while a payment one cent above interest (which amortizes) does not.

---
_Generated by [Claude Code](https://claude.ai/code)_
